tardis/interfaces/siteadapter.py

```diff
@@ -146,6 +146,13 @@ def drone_environment(
 
         return drone_environment
 
+    @property
+    def drone_heartbeat_interval(self) -> int:
```

Comment:
What just comes to my mind: What happens in case the heartbeat is `<= 0`? We should maybe also consider adding those cases to the unit tests.

Reply:
Thanks for the reminder. `0` is an allowed value. For negative values https://github.com/MatterMiners/tardis/blob/e4e35068c1c083a40ef0f0a878bd9f5c93e6545f/tardis/resources/drone.py#L105 should throw a `ValueError`. I am not happy with that behaviour, however we should generally solve this issue like already mentioned in #152. I will suggest to add tests for checking that indeed a `ValueError` is thrown.

Reply:
The behavior of `asyncio.sleep` is different from the behavior of `time.sleep` with respect to valid numbers. `time.sleep` will in fact, as you mentioned throw a `ValueError` for negative values. `asyncio.sleep` on the other hand will accept negative values and effectively treat them as 0, without any errors.

Reply:
As a provisional solution, this could check the value domain and be turned into a [`cached_propery`](https://docs.python.org/3/library/functools.html#functools.cached_property). Not sure if that's worth doing just here, though.

Reply:
`cached_property` requires Python 3.8, right? So, we would have to opt for stacking `property` decorator on top of a `caching`decorator.

Reply:
> Thanks for the reminder. `0` is an allowed value. For negative values
> 
> https://github.com/MatterMiners/tardis/blob/e4e35068c1c083a40ef0f0a878bd9f5c93e6545f/tardis/resources/drone.py#L105
> should throw a `ValueError`. I am not happy with that behaviour, however we should generally solve this issue like already mentioned in #152. I will suggest to add tests for checking that indeed a `ValueError` is thrown.

I have added a check using pydantic, that accepts only values `>=0`.